src/index.js

```diff
@@ -113,3 +113,10 @@ module.exports = window.AFRAME = {
   utils: utils,
   version: pkg.version
 };
+
+// Enter VR on `vrdisplayactivate`, to handle in-VR traversal much earlier in the lifecycle.
+// This event may also fire in other circumstances (e.g. putting on the Rift headset).
+window.addEventListener('vrdisplayactivate', function (evt) {
+  var s = window.AFRAME.scenes[0];
+  if (s.effect) { s.enterVR(); } else { setTimeout(s.enterVRBound); }
```

Comment:
I don't think it is guaranteed that in the next tick the effect is going to be defined. I'm trying to find a more robust way to do this.

Reply:
Also is not what I would not expect the event to behave following the spec. The user action policy should not apply to the next tick logic.

Reply:
If you find a more robust way, great.  I tried several things and did not find another way for gesture permission to survive.  See https://github.com/WICG/interventions/issues/12

Reply:
(More specifically, https://github.com/WICG/interventions/issues/12#issuecomment-209908148)